bcipy/io/load.py

```diff
@@ -318,3 +302,297 @@ def fast_scandir(directory_name: str, return_path: bool = True) -> List[str]:
         return [f.path for f in os.scandir(directory_name) if f.is_dir()]
 
     return [f.name for f in os.scandir(directory_name) if f.is_dir()]
+
+
+class BciPySessionTaskData:
+    """Session Task Data.
+
+    This class is used to represent a single task data session. It is used to store the
+    path to the task data, as well as the parameters and other information about the task.
+
+    /<local_path>/<user_id>/<date>/<experiment_id>/<date_time>/
+        protocol.json
+        <task_date_time>/
+            parameters.json
+            **task_data**
+
+    """
+
+    def __init__(
+            self,
+            path: str,
+            user_id: str,
+            date: str,
+            experiment_id: str,
+            date_time: str,
+            task: str,
+            run: int = 1) -> None:
+
+        self.user_id = user_id
+        self.date = date
+        self.experiment_id = experiment_id.replace('_', '')
+        self.date_time = date_time.replace("_", "").replace("-", "")
+        self.session_id = f'{self.experiment_id}{self.date_time}'
+        self.date_time = date_time
+        self.task = task
+        self.run = str(run)
+        self.path = path
+        self.parameters = self.get_parameters()
+        self.task_name = self.parameters.get('task', 'unknown').replace(' ', '')
+        self.info = {
+            'user_id': user_id,
+            'date': date,
+            'experiment_id': self.experiment_id,
+            'date_time': self.date_time,
+            'task': task,
+            'task_name': self.task_name,
+            'run': run,
+            'path': path
+        }
+
+    def get_parameters(self) -> Parameters:
+        return load_json_parameters(
+            f'{self.path}/{DEFAULT_PARAMETERS_FILENAME}',
+            value_cast=True)
+
+    def __str__(self):
+        return f'BciPySessionTaskData: {self.info=}'
+
+    def __repr__(self):
+        return f'BciPySessionTaskData: {self.info=}'
+
+
+class BciPyCollection:
+    """BciPy Data.
+
+    This class is used to represent a full BciPy data collection.
+    """
+
+    def __init__(
+            self,
+            data_directory: str,
+            experiment_id: Optional[str] = None,
+            user_id: Optional[str] = None,
+            date: Optional[str] = None,
+            date_time: Optional[str] = None,
+            excluded_tasks: Optional[List[str]] = None,
+            anonymize: bool = False) -> None:
```

Comment:
This class could use documentation on its parameters. It seems like many of these are used for filtering operations. Maybe they should be renamed with a 'filter' suffix (ex. `date_filter`, `user_id_filter`).